### PR TITLE
Courses in homepage section table

### PR DESCRIPTION
**Pull request by Bjvanminnen**
image

If your section was assigned to a course instead of a script, the name was not showing up in the SectionsTable on the teacher homepage.  This fixes that.

This also does some cleanup
- Moves some logic out of the controller and into the section model
- Renamed "course" field to "assignedTitle" to make it more clear
- Writes tests for that model
- Uses path helpers instead of url helpers
- Makes the script accessor on Section use the cache

One other thing this work revealed that this PR does not do
- Right now our Courses are not cached. We probably want them to behave similarly to our scripts cache. I plan to do that work in a follow on PR, but will likely not get to it until next week.

### CODE REVIEW

File: apps/src/templates/studioHomepages/SectionsTable.jsx [hunk 1]
```diff
@@ -77,7 +77,7 @@ const SectionsTable = React.createClass({
       PropTypes.shape({
         name: React.PropTypes.string.isRequired,
         linkToProgress: React.PropTypes.string.isRequired,
-        course: React.PropTypes.string,
+        assignedTitle: React.PropTypes.string,
         linkToCourse: React.PropTypes.string,
```
Comment (Erin007) [inline apps/src/templates/studioHomepages/SectionsTable.jsx]: Does it make sense to make this prop `linkToAssignedTitle` now?

File: dashboard/test/models/section_test.rb [hunk 2]
```diff
@@ -291,4 +291,76 @@ def verify(actual, expected)
...
   end
+
+  test 'summarize: section with a course assigned' do
+    course = create :course, name: 'somecourse'
+    section = create :section, script: nil, course: course
+
+    expected = {
+      id: section.id,
+      name: section.name,
+      linkToProgress: "//test.code.org/teacher-dashboard#/sections/#{section.id}/progress",
```
Comment (breville) [inline dashboard/test/models/section_test.rb:302]: out of curisity, how/where did `test.code.org` get introduced?

File: apps/src/templates/studioHomepages/SectionsTable.jsx (hunk 1 above)
Comment (Bjvanminnen) [reply apps/src/templates/studioHomepages/SectionsTable.jsx]: `linkToAssigned` maybe?

File: dashboard/test/models/section_test.rb (hunk 2 above)
Comment (Bjvanminnen) [reply dashboard/test/models/section_test.rb:302]: I assume because these are run as RAILS_ENV=test

**Pull request merged**